diff --git a/utils/cups-browsed.c b/utils/cups-browsed.c
index a0de0904..890442ab 100644
--- a/utils/cups-browsed.c
+++ b/utils/cups-browsed.c
@@ -6605,7 +6605,6 @@ get_printer_attributes(const char* uri)
   static const char * const pattrs[] =
     {
      "all",
-     "media-col-database"
     };
 
   /* Request printer properties via IPP to generate a PPD file for the
@@ -6618,8 +6617,11 @@ get_printer_attributes(const char* uri)
 			       host_name, sizeof(host_name),
 			       &(host_port),
 			       resource, sizeof(resource));
-  if (uri_status != HTTP_URI_OK)
+  if (uri_status != HTTP_URI_OK  &&
+  uri_status != HTTP_URI_STATUS_UNKNOWN_SCHEME){
+    debug_printf("ERROR: Invalid URI: %s",uri);
     return NULL;
+  }
   if ((http_printer =
        httpConnectEncryptShortTimeout (host_name, host_port,
 				       HTTP_ENCRYPT_IF_REQUESTED)) == NULL) {
@@ -6632,6 +6634,7 @@ get_printer_attributes(const char* uri)
   ippAddStrings(request, IPP_TAG_OPERATION, IPP_TAG_KEYWORD,
 		"requested-attributes", sizeof(pattrs) / sizeof(pattrs[0]),
 		NULL, pattrs);
+  ippSetVersion(request,1,1);
   response = cupsDoRequest(http_printer, request, resource);
 
   if (response) {
@@ -10393,7 +10396,7 @@ browse_poll_create_subscription (browsepoll_t *context, http_t *conn)
 
   request = ippNewRequest(IPP_CREATE_PRINTER_SUBSCRIPTION);
   if (context->major > 0) {
-    debug_printf("cups-browsed [BrowsePoll %s:%d]: setting IPP version %d.%d\n",
+    debug_printf("cups-browsed [BrowsePoll %s:%d]: setting   %d.%d\n",
 		 context->server, context->port, context->major,
 		 context->minor);
     ippSetVersion (request, context->major, context->minor);
